XamarinCommunityToolkit/Behaviors/UserStoppedTypingBehavior.shared.cs

```diff
@@ -0,0 +1,81 @@
+﻿using System.Threading;
+using System.Threading.Tasks;
+using System.Windows.Input;
+using Xamarin.Forms;
+
+namespace Xamarin.CommunityToolkit.Behaviors
+{
+    public class UserStoppedTypingBehavior : BaseBehavior
+    {
+        CancellationTokenSource tokenSource;
+
+        #region Bindable Properties
+
+        public static readonly BindableProperty CommandProperty = BindableProperty.Create(nameof(Command), typeof(ICommand), typeof(UserStoppedTypingBehavior));
+        public ICommand Command
+        {
+            get => (ICommand)GetValue(CommandProperty);
+            set => SetValue(CommandProperty, value);
+        }
+
+        public static readonly BindableProperty StoppedTypingThresholdProperty = BindableProperty.Create(nameof(StoppedTypingThreshold), typeof(int), typeof(UserStoppedTypingBehavior), defaultValue: 1000);
+        public int StoppedTypingThreshold
+        {
+            get => (int)GetValue(StoppedTypingThresholdProperty);
+            set => SetValue(StoppedTypingThresholdProperty, value);
+        }
+
+        public static readonly BindableProperty AutoDismissKeyboardProperty = BindableProperty.Create(nameof(AutoDismissKeyboard), typeof(bool), typeof(UserStoppedTypingBehavior), defaultValue: false);
+        public bool AutoDismissKeyboard
+        {
+            get => (bool)GetValue(AutoDismissKeyboardProperty);
+            set => SetValue(AutoDismissKeyboardProperty, value);
+        }
+
+        #endregion Bindable Properties
+
+        protected override void OnAttachedTo(View view)
+        {
+            base.OnAttachedTo(view);
+
+			if (!(view is InputView inputView)) return;
+
+			inputView.TextChanged += InputView_TextChanged;
+        }
+
+        protected override void OnDetachingFrom(View view)
+        {
+            base.OnDetachingFrom(view);
+
+			if (!(view is InputView inputView)) return;
+			
+			inputView.TextChanged -= InputView_TextChanged;
+        }
+
+        void InputView_TextChanged(object sender, TextChangedEventArgs e)
```

Comment:
OnInputViewTextChanged

But I propose you to override existing OnViewPropertyChanged. There you will be able to cast "View" and retrieve necessary property

Reply:
Hi @AndreiMisiukevich, but why would you want to override the OnViewPropertyChanged for this, as the InputView already has a specific event (TextChanged) for this? Otherwise the OnViewPropertyChanged code will be called for all property changes, not only for the Text property to change, and we need to add a check if the Text property has changed. I don't see the benefits of it and why that would be a better solution. The naming change has been made (once I commit my changes).

Reply:
in that case, you won't have to take care of unsubscribing from your event.

```csharp
protected override void OnViewPropertyChanged(object sender, PropertyChangedEventArgs e) {
    if (e.PropertyName == InputView.TextProperty.PropertyName) {
        var text = ((InputView)View).Text;
    }
}
```

but uh